src/evaluate.jl

```diff
@@ -96,43 +96,42 @@ end
 Evaluate a `HomogeneousPolynomial` polynomial using Horner's rule (hand coded)
 at `vals`.
 """
-function evaluate{T<:Number,S<:Union{Real,Complex}}(a::HomogeneousPolynomial{T},
-    vals::Array{S,1} )
-
-    numVars = get_numvars()
-    @assert length(vals) == numVars
+# function evaluate{T<:Number}(a::HomogeneousPolynomial{T}, vals::Array{T,1} )
+function evaluate{T<:Number,S<:NumberNotSeriesN}(a::HomogeneousPolynomial{T},
+        vals::Array{S,1} )
+    @assert length(vals) == get_numvars()
     R = promote_type(T,S)
+    numVars = get_numvars()
     suma = convert(TaylorN{R}, a)
 
-    for nv = 1:numVars
+    @inbounds for nv = 1:numVars
         suma = horner(suma, (nv, vals[nv]))
     end
 
     return suma.coeffs[1].coeffs[1]
 end
+evaluate(a::HomogeneousPolynomial) = zero(a.coeffs[1])
 
 """
     evaluate(a, [vals])
 
-Evaluate a `TaylorN` polynomial using Horner's rule (hand coded) at `vals`.
+Evaluate the `TaylorN` polynomial `a` using Horner's rule (hand coded) at `vals`.
 If `vals` is ommitted, it is evaluated at zero.
 """
-function evaluate{T<:Number,S<:Union{Real,Complex}}(a::TaylorN{T},
-    vals::Array{S,1} )
-
+function evaluate{T<:Number,S<:NumberNotSeriesN}(a::TaylorN{T}, vals::Array{S,1} )
+    @assert length(vals) == get_numvars()
     numVars = get_numvars()
-    @assert length(vals) == numVars
     R = promote_type(T,S)
     suma = convert(TaylorN{R}, a)
 
-    for nv = 1:numVars
+    @inbounds for nv = 1:numVars
         suma = horner(suma, (nv, vals[nv]))
     end
 
     return suma.coeffs[1].coeffs[1]
 end
 
-evaluate{T<:Number}(a::TaylorN{T}) = evaluate(a, zeros(T, get_numvars()))
+evaluate{T<:Number}(a::TaylorN{T}) = a.coeffs[1].coeffs[1]
```

Comment:
Evaluating a polynomial on zero is equivalent to the independent order zero) coefficient.